fitapp/tasks.py

```diff
@@ -46,24 +46,22 @@ def unsubscribe(*args, **kwargs):
         raise Reject(exc, requeue=False)
 
 
-
 @shared_task
-def get_time_series_data(fitbit_user, cat, resource, date=None):
+def get_time_series_data(fitbit_user, categories=[], date=None):
     """ Get the user's time series data """
 
-    try:
-        _type = TimeSeriesDataType.objects.get(category=cat, resource=resource)
-    except TimeSeriesDataType.DoesNotExist:
-        logger.exception("The resource %s in category %s doesn't exist" % (
-            resource, cat))
+    filters = {'category__in': categories} if categories else {}
+    types = TimeSeriesDataType.objects.filter(**filters)
+    if not types.exists():
+        logger.exception("Couldn't find the time series data types")
         raise Reject(sys.exc_info()[1], requeue=False)
 
     # Create a lock so we don't try to run the same task multiple times
     sdat = date.strftime('%Y-%m-%d') if date else 'ALL'
-    lock_id = '{0}-lock-{1}-{2}-{3}'.format(__name__, fitbit_user, _type, sdat)
+    cats = '-'.join('%s' % i for i in categories)
+    lock_id = '{0}-lock-{1}-{2}-{3}'.format(__name__, fitbit_user, cats, sdat)
     if not cache.add(lock_id, 'true', LOCK_EXPIRE):
```

Comment:
@brad I don't think we can use the Django cache for the lock and guarantee that it will work with the various setups that people are likely to have. For example, Django's default caching method is [local memory caching](https://docs.djangoproject.com/en/1.8/topics/cache/#local-memory-caching), which is a per-process cache. Depending on celery setup, this code can be executed by more than one process which would each have their own cache and not be able to see the locks created by the other workers.